basic_network.go

```diff
@@ -782,6 +805,51 @@ func handleTranscodeResponse(nw *BasicVideoNetwork, remotePID peer.ID, tr Transc
 	return ErrHandleMsg
 }
 
+func handleTranscodeSub(nw *BasicVideoNetwork, conn net.Conn, ts TranscodeSubMsg) error {
+	glog.Infof("%v In handleTranscodeSub; received from %v", nw.NetworkNode.ID(), conn.RemotePeer())
+	ok := nw.NetworkNode.VerifyTranscoderSig(ts.BytesForSigning(), ts.Sig, ts.StrmID)
+	if !ok {
+		// XXX fix this; need capability to check sig in calling app
+		//return errors.New("failed to verify transcoder signature")
+	}
+	// if we're the target node, open a direct cxn. otherwise, re-transmit msg
+	bcastPid, err := extractNodeID(ts.StrmID)
+	if err != nil {
+		glog.Errorf("Error extracting node id from streamID: %v", ts.StrmID)
+		return ErrHandleMsg
+	}
+	if string(bcastPid) != string(nw.NetworkNode.ID()) {
+		// we're not the target; re-transmit message
+		glog.Infof("%v Re-transmitting TranscodeSubID to peers", nw.NetworkNode.ID())
+		return handleSub(nw, TranscodeSubID, ts.StrmID, ts, conn.RemotePeer())
+	}
+	// open direct connection. first, convert multiaddrs as needed
+	mas := make([]string, len(ts.MultiAddrs))
+	if len(ts.MultiAddrs) <= 0 {
+		glog.Errorf("No multiaddrs; can't establish direct cxn to transcoder")
+		return ErrBadMultiaddrs
```

Comment:
Currently we reset the stream (similar to dropping the connection) if we return anything other than ErrHandleMsg in the handlers - not sure we should do that here. Reference: `SetupProtocol()`

Reply:
Fixed in https://github.com/livepeer/go-livepeer-basicnet/pull/32/commits/bdb52f96dab751294da9bce3fd2a5ba0751dcbfc